app/api/v2/resource/phones.rb

```diff
@@ -107,12 +117,10 @@ def validate_phone!(phone_number)
           validate_phone!(declared_params[:phone_number])
 
           phone_number = Phone.international(declared_params[:phone_number])
-          phone = current_user.phones.find_by(number: phone_number,
-                                                 code: declared_params[:verification_code])
-
-          error!({ errors: ['resource.phone.verification_invalid'] }, 404) unless phone
+          verification = Phone.verify_code(number: phone_number, code: declared_params[:verification_code])
+          error!({ errors: ['resource.phone.verification_invalid'] }, 404) unless verification.status == 'approved'
 
-          phone.update(validated_at: Time.current)
+          current_user.phones.find_by(number: phone_number).update(validated_at: Time.current)
```

Comment:
the problem is, that now you don't check if the user trying to validate his own phone.
We should have an assignment `current_user.phones.find_by(number: phone_number)` before doing action with twilio